chalice/deploy/deployer.py

```diff
@@ -280,7 +308,48 @@ def deploy(self, config, existing_resources, stage_name):
                 config, function_name, stage_name)
             deployed_values['api_handler_name'] = function_name
         deployed_values['api_handler_arn'] = function_arn
-        return deployed_values
+
+    def _deploy_auth_handlers(self, config, existing_resources, stage_name,
+                              deployed_values):
+        # type: (Config, OPT_RESOURCES, str, Dict[str, Any]) -> None
+        # The method makes the assumption that _deploy_api_handler
+        # has already been called.  As a result, it reused portions of that
+        # functions configuration:
+        auth_handlers = config.chalice_app.builtin_auth_handlers
+        if not auth_handlers:
+            return
+        for auth_config in auth_handlers:
+            self._deploy_auth_handler(
+                config, auth_config, stage_name, deployed_values)
+
+    def _deploy_auth_handler(self, config, auth_config,
+                             stage_name, deployed_values):
+        # type: (Config, app.BuiltinAuthConfig, str, Dict[str, Any]) -> None
+        api_handler_name = deployed_values['api_handler_name']
+        role_arn = self._get_or_create_lambda_role_arn(
+            config, api_handler_name)
+        zip_contents = self._osutils.get_file_contents(
+            self._packager.deployment_package_filename(config.project_dir),
+            binary=True)
+        function_name = api_handler_name + '-' + auth_config.name
+        if self._aws_client.lambda_function_exists(function_name):
+            response = self._update_lambda_function(
+                config, function_name, stage_name)
+            function_arn = response['FunctionArn']
+        else:
+            function_arn = self._aws_client.create_function(
```

Comment:
Forgot to capture this offline discussion:

In general yes, but this might end up being part of a larger refactoring to the deploy code base.  This is already going to change when I add support for configuring the auth handler lambda functions.  I'll see if I can replace consolidate when I add configuration support, but otherwise I might want to hold off for the larger deployment refactoring.